AmplifyPlugins/API/AWSAPICategoryPluginIntegrationTests/GraphQL/GraphQLWithUserPoolIntegrationTests/AuthDirective/GraphQLAuthDirectiveIntegrationTests.swift

```diff
@@ -0,0 +1,244 @@
+
+            let awsConfiguration = try TestConfigHelper.retrieveAWSConfiguration(
+                forResource: GraphQLAuthDirectiveIntegrationTests.awsconfiguration)
+            AWSInfo.configureDefaultAWSInfo(awsConfiguration)
+        } catch {
+            XCTFail("Error during setup: \(error)")
+        }
+    }
+
+    override func setUp() {
+        do {
+            AuthHelper.initializeMobileClient()
+
+            Amplify.reset()
+
+            try Amplify.add(plugin: AWSAPIPlugin())
+
+            let amplifyConfig = try TestConfigHelper.retrieveAmplifyConfiguration(
+                forResource: GraphQLAuthDirectiveIntegrationTests.amplifyConfiguration)
+            try Amplify.configure(amplifyConfig)
+
+            ModelRegistry.register(modelType: SocialNote.self)
+        } catch {
+            XCTFail("Error during setup: \(error)")
+        }
+    }
+
+    override func tearDown() {
+        Amplify.reset()
+        AuthHelper.signOut()
+    }
+
+    /// A model with operations [create, update, delete] allows the model to be read by others but not updated or
+    /// deleted. This creates a read-only model.
+    ///
+    /// - When:
+    ///    - Model restricts access to create, update, delete.
+    /// - Then:
+    ///    - Owner can create and update the model
+    ///    - Others can read the owner's model
+    ///    - Others cannot update or delete the owner's model
+    ///    - Owner can delete the model
+    func testModelIsReadOnly() {
+        AuthHelper.signIn(username: GraphQLAuthDirectiveIntegrationTests.user1,
+                          password: GraphQLAuthDirectiveIntegrationTests.password)
+        let id = UUID().uuidString
+        let content = "owner created content"
+        let ownerCreatedNoteResult = createNote(id, content: content)
+        guard case let .success(ownerCreatedNote) = ownerCreatedNoteResult else {
+            XCTFail("Owner should be able to successfully create a note")
+            return
+        }
+        let ownerReadNoteResult = queryNote(byId: ownerCreatedNote.model.id)
+        guard case let .success(ownerReadNoteOptional) = ownerReadNoteResult,
+            let ownerReadNote = ownerReadNoteOptional else {
+            XCTFail("Owner should be able to query for own note")
+            return
+        }
+        let ownerUpdateNote = SocialNote(id: ownerReadNote.model.id, content: "owner updated content", owner: nil)
+        let ownerUpdatedNoteResult = updateNote(ownerUpdateNote, version: ownerReadNote.syncMetadata.version)
+        guard case let .success(ownerUpdatedNote) = ownerUpdatedNoteResult else {
+            XCTFail("Owner should be able to update own note")
+            return
+        }
+        AuthHelper.signOut()
+        AuthHelper.signIn(username: GraphQLAuthDirectiveIntegrationTests.user2,
+                          password: GraphQLAuthDirectiveIntegrationTests.password)
+        let otherReadNoteResult = queryNote(byId: id)
```

Comment:
Technically this is cheating since we had the Id -- is it a ton more work to query by some other means?

Reply:
in a real scenario, the other would probably listQuery with a reduced selection set to display minimal info and then when someone taps on the item, they will then load the details with the id using getQuery. so someone has to choose which item they want to load. in our case, we're testing permissions to be able to get that data back, so whether we're doing a `listQuery(filter: SocialNote.keys.id == id)` or `queryNote(byId: id)`, we would still need the id to begin with